### DISCUSSION TITLE
Add config path prefixes to hls chunklist URI

### DISCUSSION THREAD

**Discussion by chris-ruecker** [OvenMediaLabs/OvenMediaEngine#1485] [Help]
**Describe the bug**
I am running OME behind a reverse proxy, and i have a rule that forwards requests to `domain.tld/llhls` to the LLHLS port of OME. 
Getting the playlist works fine, but the playlist file does not contain the /llhls prefix, so playback fails.

I would like to avoid configuring a separate subdomain for it, is there a way to add a prefix to the chunk URL?

The playlist i am getting looks like this:

```

#EXTM3U
#EXT-X-VERSION:7
#EXT-X-INDEPENDENT-SEGMENTS
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio_0",NAME="Audio_1",DEFAULT=YES,AUTOSELECT=YES,CHANNELS="2",URI="/app/stream/chunklist_1_audio_vqpZYOs4_llhls.m3u8?session=27_PZUm9XQ7"

#EXT-X-STREAM-INF:BANDWIDTH=8320000,RESOLUTION=2580x1080,FRAME-RATE=60.000,CODECS="avc1.640033,mp4a.40.2",AUDIO="audio_0"
/app/stream/chunklist_0_video_vqpZYOs4_llhls.m3u8?session=27_PZUm9XQ7
```

It would need to look like this:
```

#EXTM3U
#EXT-X-VERSION:7
#EXT-X-INDEPENDENT-SEGMENTS
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio_0",NAME="Audio_1",DEFAULT=YES,AUTOSELECT=YES,CHANNELS="2",URI="/llhls/app/stream/chunklist_1_audio_vqpZYOs4_llhls.m3u8?session=27_PZUm9XQ7"

#EXT-X-STREAM-INF:BANDWIDTH=8320000,RESOLUTION=2580x1080,FRAME-RATE=60.000,CODECS="avc1.640033,mp4a.40.2",AUDIO="audio_0"
/llhls/app/stream/chunklist_0_video_vqpZYOs4_llhls.m3u8?session=27_PZUm9XQ7
```

**To Reproduce**
1. Put ome behind a reverse proxy with a PathPrefix rule
2. Try to playback the llhls stream
3. See error

**Expected behavior**
Some way to define a prefix in the config, which is prepended to the chunklist URI.

**Server (please complete the following information):**
 - OS: Linux
 - OvenMediaEngine Version: v0.16.2

**Player (please complete the following information):**
I used the demo page for testing, Version 0.10.33

**Comment by bchah** [comment]
This was mentioned in #1357 and @getroot suggested using this secret LLHLS parameter: `<HLSChunklistPathDepth>0</HLSChunklistPathDepth>
`
Personally my workaround was to just use **/app** as the proxy endpoint, assuming LLHLS as the primary playback method. Then there's no change to the path. For proxying WebRTC traffic I use a **/ws** prefix which works fine as the playback URL does not change.

**Reply by chris-ruecker** [reply]
I have seen this discussion before, but that option did not work for me. I think the playlist ome returned was just empty, or had no chunklists defined.

I went with the /app route now, but a real solution at some point would be nice.

**Comment by getroot** [comment]
I think this is an issue that needs to be handled by the proxy. Changing Origin's path is very difficult (especially for OME).

**Comment by bchah** [comment]
Not sure of your proxy engine but if it supports regular expressions you might have luck with something like  `/(\/app\/stream/chunklist.*)/` =>`/\/llhls$1/`
